staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -341,16 +347,18 @@ func patchResource(
 	// applyAdmission is called every time GuaranteedUpdate asks for the updated object,
 	// and is given the currently persisted object and the patched object as input.
 	applyAdmission := func(ctx request.Context, patchedObject runtime.Object, currentObject runtime.Object) (runtime.Object, error) {
-		return patchedObject, admit(patchedObject, currentObject)
+		return patchedObject, mutateUpdateAdmit(patchedObject, currentObject)
 	}
-
 	updatedObjectInfo := rest.DefaultUpdatedObjectInfo(nil, applyPatch, applyAdmission)
 
+	// the patcher never creates new objects, fail closed
```

Comment:
Is this true? If update allows creation (some resources) then I would expect patch to work that way, also.

Reply:
Aligned patch and update now. So patch got the createValidation too. In theory a registry can patch some default object and use the result for creation.